pkg/proxy/userspace/proxier_test.go

```diff
@@ -234,6 +246,17 @@ func startProxier(p *Proxier, t *testing.T) {
 	p.OnEndpointsSynced()
 }
 
+// addServiceOnPortLocked is only used for testcases
+func addServiceOnPortLocked(p *Proxier, service proxy.ServicePortName, protocol v1.Protocol) (*ServiceInfo, int, error) {
```

Comment:
returning the port and using that everywhere rather than `svcInfo.proxyPort` just seems like a lot of code churn for no real advantage?